### DISCUSSION TITLE
[RESOLVED] Problems when source code is not in current folder

### DISCUSSION THREAD

**Discussion by CaledoniaProject** [github/codeql#8564] [General]
I have a test.cpp and my taint query works fine against it:

```
codeql database create cpptest -l=cpp -c="g++ test.cpp" --overwrite
```

But if I move test.cpp to somewhere else and run the database creation command:

```
codeql database create cpptest -l=cpp -c="g++ $HOME/src/test.cpp" --overwrite
```

And the same query will not yield any results. 

How can I debug this problem and figure out what's wrong?

**Comment by smowton** [comment]
This might be down to the `--source-root`, which is by default the directory from which `database create` was run. Normally alerts are restricted to code within the source root, which can be thought of as defining the boundary between code we're analysing and supporting / library code that isn't the subject of the analysis.

Suggestion: see if `--source-root=$HOME/src` fixes your problem.

**Reply by CaledoniaProject** [reply]
--source-root=$HOME/src worked for me

**Comment by CaledoniaProject** [comment]
Also, `-c="g++ ~/src/test.cpp"` will not work. Clearly codeql invokes bash on this, but why isn't `~` expanded to $HOME?

**Reply by smowton** [reply]
It doesn't invoke bash, it runs whatever command you give directly.

**Reply by CaledoniaProject** [reply]
I was wrong, $HOME was expanded when I invoke codeql, case closed.
